File: src/main/scala/temportalist/compression/main/client/ProxyClient.scala
package temportalist.compression.main.client

import java.util

import net.minecraft.block.state.IBlockState
import net.minecraft.client.Minecraft
import net.minecraft.client.gui.GuiScreen
import net.minecraft.client.renderer.block.model.ModelResourceLocation
import net.minecraft.client.renderer.block.statemap.StateMapperBase
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.tileentity.TileEntity
import net.minecraft.world.World
import net.minecraftforge.client.model.{ModelLoader, ModelLoaderRegistry}
import net.minecraftforge.fml.client.IModGuiFactory
import net.minecraftforge.fml.client.IModGuiFactory.{RuntimeOptionCategoryElement, RuntimeOptionGuiHandler}
import temportalist.compression.main.client.model.ModelLoaderCompressed
import temportalist.compression.main.common.init.{ModBlocks, ModItems}
import temportalist.compression.main.common.{Compression, ProxyCommon}

/**
  *
  * Created by TheTemportalist on 4/14/2016.
  *
  * @author TheTemportalist
  */
class ProxyClient extends ProxyCommon with IModGuiFactory {

	override def preInit(): Unit = {

		ModelLoaderRegistry.registerLoader(new ModelLoaderCompressed())

		ModelLoader.setCustomModelResourceLocation(ModItems.item, 0, ModelLoaderCompressed.fakeRL)
		ModelLoader.setCustomModelResourceLocation(ModBlocks.blockItem, 0, ModelLoaderCompressed.fakeRL)
		ModelLoader.setCustomStateMapper(ModBlocks.block, new StateMapperBase {
			override protected def getModelResourceLocation(
					state: IBlockState): ModelResourceLocation = ModelLoaderCompressed.fakeRL
		})

		for (slot <- ModItems.armorTypes) {
			val location = new ModelResourceLocation(
				Compression.getModId+ ":ItemDenseArmor", "part=" + slot.getName
			)
			ModelLoader.setCustomModelResourceLocation(
				ModItems.leatherDenseArmor(slot.getIndex), 0, location)
		}

	}

	override def register(): Unit = {

	}

	override def getClientElement(ID: Int, player: EntityPlayer, world: World,
			x: Int, y: Int, z: Int, tileEntity: TileEntity): AnyRef = {
		null
	}

	override def mainConfigGuiClass(): Class[_ <: GuiScreen] = classOf[GuiConfig]

	override def initialize(minecraftInstance: Minecraft): Unit = {}

	override def runtimeGuiCategories(): util.Set[RuntimeOptionCategoryElement] = null

	override def getHandlerFor(element: RuntimeOptionCategoryElement): RuntimeOptionGuiHandler = null

}

File: src/main/java/com/temportalist/compression/client/ProxyClient.java
package com.temportalist.compression.client;

import com.google.common.collect.ImmutableList;
import com.temportalist.compression.common.Compression;
import com.temportalist.compression.common.ProxyCommon;
import com.temportalist.compression.common.blocks.BlockBase;
import com.temportalist.compression.common.blocks.TileCompressor;
import com.temportalist.compression.common.init.ModBlocks;
import com.temportalist.compression.common.init.ModItems;
import net.minecraft.block.state.BlockStateContainer;
import net.minecraft.block.state.IBlockState;
import net.minecraft.client.renderer.block.model.ModelResourceLocation;
import net.minecraft.client.renderer.block.statemap.DefaultStateMapper;
import net.minecraft.client.renderer.block.statemap.StateMapperBase;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.item.ItemBlock;
import net.minecraft.util.ResourceLocation;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.World;
import net.minecraftforge.client.event.ModelRegistryEvent;
import net.minecraftforge.client.model.ModelLoader;
import net.minecraftforge.client.model.ModelLoaderRegistry;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
import net.minecraftforge.fml.common.network.NetworkRegistry;
import net.minecraftforge.fml.relauncher.Side;

import javax.annotation.Nullable;

@Mod.EventBusSubscriber(Side.CLIENT)
public class ProxyClient extends ProxyCommon {

    @Override
    public void initPre(FMLPreInitializationEvent e) {
        super.initPre(e);

        NetworkRegistry.INSTANCE.registerGuiHandler(Compression.main, this);

    }

    @SubscribeEvent
    public static void registerModels(ModelRegistryEvent event) {

        ModelLoaderRegistry.registerLoader(new ModelLoaderCompressed());

        ModelLoader.setCustomModelResourceLocation(ModBlocks.compressed.item, 0, ModelLoaderCompressed.fakeRL);
        ModelLoader.setCustomStateMapper(ModBlocks.compressed, new StateMapperBase() {

            @Override
            protected ModelResourceLocation getModelResourceLocation (IBlockState state){
                return ModelLoaderCompressed.fakeRL;
            }

        });
        ModelLoader.setCustomModelResourceLocation(ModItems.compressed, 0, ModelLoaderCompressed.fakeRL);

        ModelLoader.setCustomModelResourceLocation(ModBlocks.compressor.item, 0, new ModelResourceLocation(
                new ResourceLocation(Compression.MOD_ID, "compressor"),  "compress"
        ));
        /*
        ModelLoader.setCustomModelResourceLocation(ModBlocks.compressor.item, 1, new ModelResourceLocation(
                new ResourceLocation(Compression.MOD_ID, "compressor"),  "decompress"
        ));
        //*/

    }

    @Nullable
    @Override
    public Object getClientGuiElement(int ID, EntityPlayer player, World world, int x, int y, int z) {
        if (ID == Compression.guidIdCompressor)
        {
            return new GuiCompressor(player.inventory, (TileCompressor)world.getTileEntity(new BlockPos(x, y, z)));
        }
        return null;
    }
}
